### DISCUSSION TITLE
Inconsistencies in the cross functions

### DISCUSSION THREAD

**Discussion by janaobsteter** [HighlanderLab/SIMplyBee#282] [Q&A] [answered]
1) The basic cross function is `crossVirginQueen`. Here, we provide `drones` and `nFathers`. If we have:
- 1 virgin queen, we do:` fathers <- selectInd(pop = drones, nInd = n, use = "rand")`
- multiple virgin queens, we do:   `fatherGroups <- pullDroneGroupsFromDCA(DCA = drones,n = nVirginQueen,nFathers = nFathers)`

Note that the first one doesn't "kill" the drones, while the second one does.

2) In the `crossColony`, we don't select any fathers, but instead, just pass the `drones` to `crossVirginQueen`.

3) In the `crossColonies`, we again sample father groups, which is then again done by `crossColony` that we call in here
`fatherGroups <- pullDroneGroupsFromDCA(DCA = drones, n = nCol, nFathers = nFathers, removeFathers = removeFathers)`
Also, note that here we pass the argument `removeFathers`, where in the crossVirginQueen we don't.

Almost all of this will be fixed by collapsing the functions, but just so that we are aware of what is hapenning.

**Comment by gregorgorjanc** [comment]
Thanks for raising this @janaobsteter . It’s a bit hard to follow since you’re not showing full examples.

**Comment by janaobsteter** [comment]
Yeah, I know - but I will assign myself to collapsing this functions and solve this!

**Comment by janaobsteter** [comment]
I moved this to the discussion since I feel we need to discuss on it.
Currently, we have three cross functions - `crossVirginQueens()`, `crossColony()`, `crossColonies()`. We agreed on collapsing `crossColony() `and `crossColonies()`, but what about the `crossVirginQueens()`?

The problem is that all of them use "drones" argument, with is basically a DCA, and then internally select fathers. So, they are actually doing two things at once (which also means things are duplicated). The problem also is, that they use different functions for selecting fathers - `selectInd` in `crossVirginQueens()` and `pullDroneGroupsFromDCA` in other two.

So our options are:
- to keep `crossVirginQueens()`, but take father selection out. The question here then is, should we allow for mating multiple virgin queens at once (with user supplying father packages)?
- to collapse all three functions into one

Thoughts?

**Comment by gregorgorjanc** [comment]
As discussed in the meeting, we probably should collapse and combine and get the function do the same thing for the three cases. Then we need what else we need for specific cases.

**Comment by janaobsteter** [comment]
Ok - I will also take father selection out of the function - for now! And then we can see whether it still makes sense to have to in there.

**Comment by gregorgorjanc** [comment]
@janaobsteter I believe this discussion should be answered given your work on crossing function, right?

**Comment by janaobsteter** [comment, marked as answer]
SIMplyBee now includes functions for open and controlled mating - cross() functions accepts either a population of drones (the fathers) OR a could of drones and a correspodning cross plan.
